inspirehep/utils/url.py

```diff
@@ -65,3 +69,17 @@ def is_pdf_link(url):
     correct_magic_number = magic_number.startswith('%PDF')
 
     return correct_magic_number
+
+
+def retrieve_uri(uri, outdir=None):
+    """Retrieves the given uri and stores it in a temporary file."""
+    fd, local_file = tempfile.mkstemp(prefix='inspire', dir=outdir)
```

Comment:
You can use https://docs.python.org/2/library/tempfile.html#tempfile.NamedTemporaryFile with `deleted=False` which is slightly easier.

Reply:
This and the next will be solved with:

```
  7 def retrieve_uri(uri, outdir=None):                                              
  6     """Retrieves the given uri and stores it in a temporary file."""             
  5     fd, local_file = tempfile.mkstemp(prefix='inspire', dir=outdir)              
  4     os.close(fd)                                                                 
  3                                                                                  
  2     fs.copy.copy_file(uri, local_file)                                                   
  1                                                                                  
81      return local_file 
```

as the copy_file requires a string or fs.FS instance, not an open file (will open it with whatever opener is needed for it).
is that ok? I've checked the code underneath, and it uses chunks of:
```
In [1]: import io

In [2]: io.DEFAULT_BUFFER_SIZE
Out[2]: 8192
```

Reply:
dammit... this requires fs>=2, but it's capped by invenio-files-rest, see https://github.com/inveniosoftware/invenio-files-rest/issues/130

Reply:
You can do this with PyFSFileStorage which has all the support for chunking,  size limit checking etc:

```python
>>> import os
>>> import tempfile
>>> import requests
>>> from invenio_files_rest.storage import PyFSFileStorage
>>> fd, local_file = tempfile.mkstemp(prefix='inspire')
>>> os.close(fd)
>>> dst = PyFSFileStorage(local_file)
>>> dst.save(requests.get(uri, stream=True).raw, size_limit=1*1024*1024, chunk_size=2*1024*1024)
```

Reply:
For local disk 8kb chunk size is ok....for EOS you definitely want to go to MB size for best performance.

Reply:
Thanks for the tip, though I think it's a bit premature to strive for performance (and probably it's at the pyfs level where we want to handle each different fs type tweaking, here we know one of the files is local, but the other we are not sure).